### PR TITLE
Drop `Storage` abstraction, revise fs cleanup logic for round reset

### LINKED ISSUE #360: Drop `Storage` abstraction, and work directly with directories on the filesystem
The abstraction was originally there to help us switch backends if we needed to, but it restricts us in what we can work with when it comes to using the Disk storage like we currently do - we can't work with directories or anything, as everything needs to be done through locators.

### PR DESCRIPTION
**Pull request by julesdesmit**
Fixes #360

### CODE REVIEW

**Comment by ibaryshnikov** [comment]
Since it also touches a lot of files, I would prefer to not merge the storage refactoring before we merge #363, otherwise it may eliminate the work I've put into fixing tests.
On the other hand, it would be nice to have the round reset as a standalone tool, with an option to check the round integrity. With such a tool we would be able to stop the coordinator, do the reset, and start the coordinator again.

File: phase1-coordinator/src/storage/disk.rs
Comment (ibaryshnikov) [inline phase1-coordinator/src/storage/disk.rs:56]: I really like this approach. Resetting the round looks like a well defined problem after checking the folder structure.

File: phase1-coordinator/src/objects/round.rs
Comment (kellpossible) [inline phase1-coordinator/src/objects/round.rs:1140]: nice to see lots of code getting deleted :) despite having worked so hard on this!

File: phase1-coordinator/src/coordinator.rs
Comment (kellpossible) [inline phase1-coordinator/src/coordinator.rs:2459]: I think it would be good to represent `clear_round_files` as a storage action, returned by `round.reset()` instead of calling it directly here. That better fits with the existing pattern and improves testability.

**Review by ibaryshnikov — APPROVED** [review]

**Comment by kellpossible** [comment]
@julesdesmit  I guess you're good to merge this one?

**Pull request merged**
